Modularize parsing NCCL_BLOCKING_WAIT in ProcessGroupNCCL

osalpekar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#41076 Modularize parsing NCCL_BLOCKING_WAIT in ProcessGroupNCCL**

Modularizes Parsing the NCCL_BLOCKING_WAIT environment variable in the ProcessGroupNCCL Constructor.

Differential Revision: [D22401225](https://our.internmc.facebook.com/intern/diff/D22401225/)

osalpekar committed:
Modularize parsing NCCL_BLOCKING_WAIT in ProcessGroupNCCL

Modularizes Parsing the NCCL_BLOCKING_WAIT environment variable in the ProcessGroupNCCL Constructor.

Differential Revision: [D22401225](https://our.internmc.facebook.com/intern/diff/D22401225/)

[ghstack-poisoned]

osalpekar commented on torch/lib/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -271,19 +286,8 @@ ProcessGroupNCCL::ProcessGroupNCCL(
       ncclCommCounter_(0),
       terminateWatchdog_(false),
       opTimeout_(opTimeout) {
```
We can alternately set `blockingWait_` from the `ProcessGroupNCCL` ctor initializer list and handle all exception cases in the helper - would that be better style-wise?

jiayisuse replied:
Current solution looks good to me. Initialization list tends to do trivial initialization.

jiayisuse commented:
Seems CI fails to download cuda pkgs
```
Failed to fetch https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/Packages  Writing more data than expected (1196016 > 1190062)
```

osalpekar commented:
> Seems CI fails to download cuda pkgs
> 
> ```
> Failed to fetch https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/Packages  Writing more data than expected (1196016 > 1190062)
> ```

I'll try rebasing, seems like some flakiness in the CI

osalpekar committed:
Update on "Modularize parsing NCCL_BLOCKING_WAIT in ProcessGroupNCCL"


Modularizes Parsing the NCCL_BLOCKING_WAIT environment variable in the ProcessGroupNCCL Constructor.

Differential Revision: [D22401225](https://our.internmc.facebook.com/intern/diff/D22401225/)

[ghstack-poisoned]
